### PR TITLE
EOS-13937 : Rules for consistent error response format

### PR DESCRIPTION
**Pull request by SeagateChaDeepak**
* JSON Error Response Schema :

To be consistent in returning errors(when using REST API) with other components
such as S3, we will follow the error format that S3 currently uses.

Link :
https://github.com/Seagate/cortx-fs/edit/EOS-13937/doc/RESTAPI_Guidelines.md

Signed-off-by: Deepak Mahale <deepak.mahale@seagate.com>

### CODE REVIEW

File: doc/RESTAPI_Guidelines.md [hunk 1]
```diff
@@ -122,3 +122,52 @@ the key, its better to either create a new key for that purpose or expose a new
...
+To be consistent in returning errors(when using REST API) with other components
+such as S3, we will follow the error format that S3 currently uses 
+(Schema shown below).
+
+<details>
+<summary>Schema</summary>
+<pre>
+
+>{
+>  "$schema": "http://json-schema.org/draft-04/schema#",
```
Comment (Shreya-18) [inline doc/RESTAPI_Guidelines.md:137]: One suggestion : the indentation of this json object is not seen when we see the actual document. Which makes it difficult to read.  Can we change the format from `>` to ` ``` `  ?

Comment (SeagateChaDeepak) [reply doc/RESTAPI_Guidelines.md]: Done

**Review by Shreya-18 — APPROVED** [review]
LGTM, +1

**Review by s-arya — APPROVED** [review]
LGTM

**Pull request merged**